### PR TITLE
Consistently render links to the documentation

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we render all links to Rally's documentation
consistently with a newly introduced function `doc_link` that can also
handle paths to specific pages in the documentation.

### CODE REVIEW

File: esrally/rally.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def positive_number(v):
 
     parser = argparse.ArgumentParser(prog=PROGRAM_NAME,
                                      description=BANNER + "\n\n You Know, for Benchmarking Elasticsearch.",
-                                     epilog="Find out more about Rally at %s" % console.format.link(DOC_LINK),
+                                     epilog="Find out more about Rally at %s" % console.format.link(doc_link()),
```
Comment (dliappis) [inline esrally/rally.py]: Given the opportunity you could replace `%s` and `%` here with `format()` for standardization.

File: esrally/rally.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 import uuid
 
 from esrally import version, actor, config, paths, racecontrol, reporter, metrics, track, chart_generator, exceptions, log
-from esrally import PROGRAM_NAME, DOC_LINK, BANNER, SKULL, check_python_version
+from esrally import PROGRAM_NAME, BANNER, SKULL, doc_link, check_python_version
```
Comment (dliappis) [inline esrally/rally.py]: nit: this is just me, normally I try to have these in alphabetical order (easier said than done).

File: esrally/__init__.py [hunk 3]
```diff
@@ -17,6 +17,7 @@
 
 import os
 import sys
+import urllib
```
Comment (dliappis) [inline esrally/__init__.py]: Beyond the scope of this PR, but should we slowly settle on having alphabetically sorted imports?

File: esrally/racecontrol.py [hunk 4]
```diff
@@ -23,8 +23,8 @@
 import thespian.actors
 import urllib
 
-from esrally import actor, config, exceptions, track, driver, mechanic, reporter, metrics, time, DOC_LINK, PROGRAM_NAME
-from esrally.utils import console, convert, opts
+from esrally import actor, config, exceptions, track, driver, mechanic, reporter, metrics, doc_link, PROGRAM_NAME
```
Comment (dliappis) [inline esrally/racecontrol.py]: Same comment here, there are _almost_ sorted, so we could really have them sorted.

File: esrally/rallyd.py [hunk 5]
```diff
@@ -20,7 +20,7 @@
 import logging
 import argparse
 
-from esrally import actor, version, exceptions, DOC_LINK, BANNER, PROGRAM_NAME, check_python_version, log
+from esrally import actor, version, exceptions, doc_link, BANNER, PROGRAM_NAME, check_python_version, log
```
Comment (dliappis) [inline esrally/rallyd.py]: Same comment here, there are _almost_ sorted, so we could really have them sorted.

File: esrally/rallyd.py [hunk 6]
```diff
@@ -76,7 +76,7 @@ def main():
 
     parser = argparse.ArgumentParser(prog=PROGRAM_NAME,
                                      description=BANNER + "\n\n Rally daemon to support remote benchmarks",
-                                     epilog="Find out more about Rally at %s" % console.format.link(DOC_LINK),
+                                     epilog="Find out more about Rally at %s" % console.format.link(doc_link()),
```
Comment (dliappis) [inline esrally/rallyd.py]: Perhaps take the opportunity to use `format()` here for consistency?

File: tests/racecontrol_test.py [hunk 7]
```diff
@@ -17,11 +17,9 @@
 
 import unittest.mock as mock
 import os
-import random
 from unittest import TestCase
 
-from esrally import racecontrol, config, exceptions, DOC_LINK
-from esrally.utils import console
+from esrally import racecontrol, config, exceptions
```
Comment (dliappis) [inline tests/racecontrol_test.py]: Same comment for alphabetical order.

File: esrally/rally.py (hunk 2 above)
Comment (danielmitterdorfer) [reply esrally/rally.py]: I tried to keep constants and functions separately that's why I moved the `doc_link`

File: esrally/__init__.py (hunk 3 above)
Comment (danielmitterdorfer) [reply esrally/__init__.py]: I *try* to keep imports somewhat logically grouped so far but I don't mind too much. I am just not certain how we would enforce this on an existing code base. But let's discuss this separately.

File: esrally/rally.py (hunk 1 above)
Comment (danielmitterdorfer) [reply esrally/rally.py]: Good point. Will do.

**Comment by danielmitterdorfer** [comment]
Thanks for the review and suggestions. I've addressed all of them in a follow-up commit.

**Pull request merged**